src/main/java/java_echo_server/Client.java

```diff
@@ -0,0 +1,28 @@
+package java_echo_server;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.net.Socket;
+
+public class Client {
+    private final BufferedReader inputStream;
+    private final WriterWrapper outputStream;
+
+    public Client(BufferedReader input, WriterWrapper output) {
+        inputStream = input;
```

Comment:
A bit of a nitpick, but conventionally we use the same variable names as the constructor parameters and use the `this` keyword to specify instance properties.

```java
public Client(final BufferedReader input, final WriterWrapper output) {
    this.input = input;
    this.output = output;
}
```